Added support for decide function

francislaus opened the pull request:
Added the implementation of the decide function and its corresponding test cases. This will close #35

francislaus committed:
Started the implementation of decide for #35

Added the implementation of the function decide and the corresponding
test cases as far as possible

francislaus committed:
Fix test cases for #35

Implemented the missing test cases for the function decide. The test
cases are only partially working

francislaus committed:
Merge branch 'master' into issue35

Resolve conflict with master to merge and fix #35

francislaus committed:
Added support for decide function to fix #35

Finished the implementation of the decide function and its corresponding
test cases

apeinot reviewed (changes requested):
The code and tests are globally fine. A few things to fix:
- In decide(): Final launch / no launch decision encoded as ”YES”, ”NO” on the standard output. (from the problem statement)
-In test: there are small mistakes in the comments of the functions

With this changes, it should fix #35

Samuelhertzberg commented:
There is a small typo in the comments of the test. It say the test should return true even though it asserts it to false. Additionally it says "// first and second test case MISSING" right after first and second test case :S

francislaus committed:
Fixed problems in the decide process for #35

Improved the implementation of the decide function so that it
corresponds with the problem statement. The description of the test
cases was also adopted
